src/python/WMCore/JobSplitting/JobFactory.py

```diff
@@ -27,11 +24,11 @@ class JobFactory(WMObject):
     def __init__(self,
                  package='WMCore.DataStructs',
                  subscription=None,
-                 generators=[],
+                 generators=None,
                  limit = 0):
         self.package = package
         self.subscription  = subscription
-        self.generators    = generators
+        self.generators    = generators if generators else []
```

Comment:
They are different. The new code creates a new empty list every time the function is called. With the [] in the call definition that is not the case.